meshio/med_io.py

```diff
@@ -180,52 +221,86 @@ def write(filename, mesh, add_global_ids=True):
     coo.attrs.create("CGT", 1)
     coo.attrs.create("NBR", len(mesh.points))
 
+    # Point tags
+    if "point_tags" in mesh.point_data:  # works only for med -> med
+        fam = noe_group.create_dataset("FAM", data=mesh.point_data["point_tags"])
+        fam.attrs.create("CGT", 1)
+        fam.attrs.create("NBR", len(mesh.points))
+
     # Cells (mailles in french)
     mai_group = ts.create_group("MAI")
     mai_group.attrs.create("CGT", 1)
-    for key, med_type in meshio_to_med_type.items():
-        if key in mesh.cells:
-            mai = mai_group.create_group(med_type)
-            mai.attrs.create("CGT", 1)
-            mai.attrs.create("CGS", 1)
-            mai.attrs.create("PFL", pfl.encode("ascii"))
-            nod = mai.create_dataset("NOD", data=mesh.cells[key].T.flatten() + 1)
-            nod.attrs.create("CGT", 1)
-            nod.attrs.create("NBR", len(mesh.cells[key]))
-
-    # Subgroups (familles in french)
+    for cell_type, cells in mesh.cells.items():
+        med_type = meshio_to_med_type[cell_type]
+        mai = mai_group.create_group(med_type)
+        mai.attrs.create("CGT", 1)
+        mai.attrs.create("CGS", 1)
+        mai.attrs.create("PFL", pfl.encode("ascii"))
+        nod = mai.create_dataset("NOD", data=cells.T.flatten() + 1)
+        nod.attrs.create("CGT", 1)
+        nod.attrs.create("NBR", len(cells))
+
+        # Cell tags
+        if cell_type in mesh.cell_data:
+            if "cell_tags" in mesh.cell_data[cell_type]:  # works only for med -> med
+                fam = mai.create_dataset(
+                    "FAM", data=mesh.cell_data[cell_type]["cell_tags"]
+                )
+                fam.attrs.create("CGT", 1)
+                fam.attrs.create("NBR", len(cells))
+
+    # Information about point and cell sets (familles in french)
     fas = f.create_group("FAS")
     fm = fas.create_group(mesh_name)
     fz = fm.create_group("FAMILLE_ZERO")  # must be defined in any case
     fz.attrs.create("NUM", 0)
 
+    # For point tags
+    try:
+        if len(mesh.point_tags) > 0:
+            noeud = fm.create_group("NOEUD")
```

Comment:
Let's not use French variable names ourselves. `nodes` would be fine. (Or `knoten`, since I'm from Germany. Your choice. :smile:)

Reply:
Ok, I will let Molière speak Shakespeare (or Goethe).